docs/resources/client/models/bakedmodel.md

```diff
@@ -0,0 +1,117 @@
+| `FIRST_PERSON_LEFT_HAND`  | `"firstperson_lefthand"`  | Left hand in first person                                                                                        |
+| `HEAD`                    | `"head"`                  | When in a player's head armor slot (often only achievable via commands)                                          |
+| `GUI`                     | `"gui"`                   | Inventories, player hotbar                                                                                       |
+| `GROUND`                  | `"ground"`                | Dropped items; note that the rotation of the dropped item is handled by the dropped item renderer, not the model |
+| `FIXED`                   | `"fixed"`                 | Item frames                                                                                                      |
+
+## `ItemOverrides`
+
+`ItemOverrides` is a class that provides a way for baked models to process the state of an [`ItemStack`][itemstack] and return a new baked model through the `#resolve` method. `#resolve` has five parameters:
+
+- A `BakedModel`: The original model.
+- An `ItemStack`: The affected item stack.
+- A `ClientLevel`: The level the model is being rendered in. This should only be used for querying the level, not mutating it in any way. May be null.
+- A `LivingEntity`: The entity the model is rendered on. May be null, e.g. when rendering from a [block entity renderer][ber].
+- An `int`: A seed for randomizing.
+
+`ItemOverrides` also hold the model's override options as `BakedOverride`s. An object of `BakedOverride` is an in-code representation of a model's [`overrides`][overrides] block. It can be used by baked models to return different models depending on its contents. A list of all `BakedOverride`s of an `ItemOverrides` instance can be retrieved through `ItemOverrides#getOverrides()`.
+
+## `BakedModelWrapper`
+
+A `BakedModelWrapper` can be used to modify an already existing `BakedModel`. `BakedModelWrapper` is a subclass of `BakedModel` that accepts another `BakedModel` (the "original" model) in the constructor and by default redirects all methods to the original model. Your implementation can then override only select methods, like so:
+
+```java
+// The generic parameter may optionally be a more specific subclass of BakedModel.
+// If it is, the constructor parameter must match that type.
+public class MyBakedModelWrapper extends BakedModelWrapper<BakedModel> {
+    // Pass the original model to super.
+    public MyBakedModelWrapper(BakedModel originalModel) {
+        super(originalModel);
+    }
+    
+    // Override whatever methods you want here. You may also access originalModel if needed.
+}
+```
+
+After writing your model wrapper class, you must apply the wrappers to the models it should affect. Do so in a [client-side][sides] [event handler][event] for `ModelEvent.ModifyBakingResult`:
```

Comment:
I'm not sure we should be encouraging wrapping this prominently in the documentation. I have seen several mods abuse it when they should just use a custom model loader for their models instead.

Reply:
Then it's probably better to list when it makes to use a wrapper or a custom model loader.